### PR TITLE
Improve error messaging around bootstrap extension provisioning

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Make the extension transient error message in CAPZ controller logs a little clearer. Thanks @jsturtevant for the nudge.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Improve error messaging around bootstrap extension provisioning
```

### CODE REVIEW

**Review by mboersma — APPROVED** [review]
lgtm

These are more clear and informative. 👍🏻

**Review by fiunchinho — APPROVED** [review]

File: azure/errors.go [hunk 1]
```diff
@@ -89,11 +89,9 @@ func (t ReconcileError) Error() string {
 	}
 	switch t.errorType {
 	case TransientErrorType:
-		return fmt.Sprintf("reconcile error occurred that can be recovered. Object will be requeued after %s "+
-			"The actual error is: %s", t.requestAfter.String(), errStr)
+		return fmt.Sprintf("transient reconcile error occurred: %s. Object will be requeued after %s", errStr, t.requestAfter.String())
 	case TerminalErrorType:
-		return fmt.Sprintf("reconcile error occurred that cannot be recovered. Object will not be requeued. "+
-			"The actual error is: %s", errStr)
+		return fmt.Sprintf("reconcile error that cannot be recovered occurred: %s. Object will not be requeued", errStr)
```
Comment (nader-ziada) [inline azure/errors.go:94]: The new error messages doesn't match the expected message for tests `TestReconcileVMSS ` and `TestReconcileVM `

**Review by nader-ziada — CHANGES_REQUESTED** [review]

File: azure/errors.go (hunk 1 above)
Comment (CecileRobertMichon) [reply azure/errors.go:94]: fixed

**Comment by CecileRobertMichon** [comment]
/retest

ipv6 flake

**Comment by CecileRobertMichon** [comment]
/retest 

prow flake

**Pull request merged**